osu.Game/Updater/UpdateManager.cs

```diff
@@ -35,24 +37,37 @@ protected override void LoadComplete()
         {
             base.LoadComplete();
 
+            Schedule(() => Task.Run(CheckForUpdateAsync));
+
+            // debug / local compilations will reset to a non-release string.
+            // can be useful to check when an install has transitioned between release and otherwise (see OsuConfigManager's migrations).
+            config.Set(OsuSetting.Version, game.Version);
+        }
+
+        public async Task CheckForUpdateAsync()
+        {
+            if (!CanCheckForUpdate)
+                return;
+
+            await InternalCheckForUpdateAsync();
+        }
+
+        protected virtual Task InternalCheckForUpdateAsync()
+        {
+            // Query last version only *once*, so the user can re-check for updates, in case they closed the notification or else.
+            lastVersion ??= config.Get<string>(OsuSetting.Version);
+
             var version = game.Version;
-            var lastVersion = config.Get<string>(OsuSetting.Version);
 
-            if (game.IsDeployedBuild && version != lastVersion)
+            if (version != lastVersion)
             {
                 // only show a notification if we've previously saved a version to the config file (ie. not the first run).
                 if (!string.IsNullOrEmpty(lastVersion))
                     Notifications.Post(new UpdateCompleteNotification(version));
             }
 
-            // debug / local compilations will reset to a non-release string.
-            // can be useful to check when an install has transitioned between release and otherwise (see OsuConfigManager's migrations).
-            config.Set(OsuSetting.Version, version);
-        }
-
-        public virtual void CheckForUpdate()
-        {
-            Logger.Log("CheckForUpdate was called on the base class (UpdateManager)", LoggingTarget.Information);
+            // we aren't doing any async in this method, so we return a completed task instead.
+            return Task.CompletedTask;
```

Comment:
what is this doing here, should stay in `LoadComplete`?

Reply:
You highlighted two methods, what do you mean by "this"?

Reply:
Another weird thing by GItHub, anyways why is the version finalizing logic inside `InternalCheckForUpdateAsync`, it should stay where it is at `LoadComplete`?

Reply:
I mixed up the "Check what's new" code as another checking code, I'll be reverting this the next push.